src/OpenLoco/src/Vehicles/Vehicle.cpp

```diff
@@ -686,6 +686,88 @@ namespace OpenLoco::Vehicles
         return AirportObjectFlags::acceptsHeavyPlanes;
     }
 
+    // 0x004AFFF3
+    VehicleBogie* VehicleBogie::flipCar()
+    {
+        Vehicle train(head);
+        VehicleBase* precedingCarBody = train.veh2;
+        while (precedingCarBody->nextVehicleComponent()->asBase<VehicleBogie>() != this)
+        {
+            precedingCarBody = precedingCarBody->nextVehicleComponent();
+        }
+
+        VehicleBase* ptr = this; // lvalue required
+        CarComponent oldFirstComponent(ptr);
+        ptr = this; // CarComponent constructor changes value
+        Car car(ptr);
+
+        sfl::static_vector<CarComponent, VehicleObject::kMaxCarComponents> components;
+        for (CarComponent& component : car)
+        {
+            auto frontSprite = component.front->objectSpriteType;
+            component.front->objectSpriteType = component.back->objectSpriteType;
```

Comment:
you could std::swap to avoid having to have a frontSprite var.
`std::swap(component.front->objectSpriteType, component.back->objectSpriteType);`